vignettes/design-principles.Rmd

```diff
@@ -0,0 +1,42 @@
+---
+title: "Design Principles for linelist"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Design Principles for linelist}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+This vignette outlines the design decisions that have been taken during the development of the `{linest}` R package, and provides some of the reasoning, and possible pros and cons of each decision.
+
+This document is primarily intended to be read by those interested in understanding the code within the package and for potential package contributors.
+
+## Scope
+
+linelist provides a lightweight layer to add tags to data.frame columns. This allows:
+
+- column identification without renaming
+- extra feature for the tagged columns, such as the ability to warn when a tagged column is dropped, or when its data type is incompatible with the expected one.
+
+## Input/Output/Interoperability
+
+Because of its scope, linelist is intended to provide maximum compatibility with data.frames, or packages defining subclasses of data.frames.
```

Comment:
I assume the position of {linelist} is that any difference of behaviour between `<data.frame>` subclasses should be understood by the user and is not the concern of {linelist} which is just wrapping? E.g. https://tibble.tidyverse.org/articles/wip/subset.html.

Reply:
Yes, it's now explicitly called out via [`76926a7` (#112)](https://github.com/epiverse-trace/linelist/pull/112/commits/76926a7e9ec471a423d4660ab1bd5d3e4ae3e24d).